keyboards/crkbd/keymaps/mcrown/oled.c

```diff
@@ -0,0 +1,384 @@
+ *
+ *  If any argument is invalid, the function has no effect.
+ *
+ *  @param oled_rotation_t rotation
+ *  @return rotation of the display.
+ */
+oled_rotation_t oled_init_user(oled_rotation_t rotation)
+{
+    oled_rotation_t oled_rot=OLED_ROTATION_180;
+
+    if(TRUE==is_master)
+    {
+#ifdef OLED_VERTICAL
+        oled_rot=OLED_ROTATION_270;
+#else
+        oled_rot=OLED_ROTATION_0;
+#endif
+    }
+    return oled_rot;
+}
+
+/** @brief renders the keylog string and display it. This function also toggles the cursor.
+ *
+ *  If any argument is invalid, the function has no effect.
+ *
+ *  @param void.
+ *  @return void.
+ */
+static void render_keylogger_status(void)
+{
+    static bool cursor_f=TRUE;
+
+    if(timer_elapsed32(cursor_oled_timer) > 300)
+    {
+        cursor_oled_timer = timer_read32();
+        cursor_f=TOGGLE_BOOL_VAR(cursor_f);
+    }
+    oled_write_P(PSTR("\n>:"), FALSE);
+    if(current_cursor_pos>(KEYLOG_LEN-1))
+    {
+        current_cursor_pos=0;
+        memset(keylog_str, ' ', sizeof(char)*KEYLOG_EOL_LEN);
+        /* Here the EOL is to clear with white spaces all the keylog area */
+        keylog_str[KEYLOG_EOL_LEN-1] = '\0';
+        oled_write(keylog_str, FALSE);
+        /* Reset EOL to the begining of the keylog string */
+        keylog_str[0] = '\0';
+    }
+    oled_write(keylog_str, FALSE);
+    oled_write_char(last_c, cursor_f);
+
+}
+
+/** @brief displays the current active layout.
+ *
+ *  If any argument is invalid, the function has no effect.
+ *
+ *  @param void.
+ *  @return void.
+ */
+void render_layout_state(void)
+{
+#ifdef OLED_VERTICAL
+    oled_write_P(PSTR("Lyt:\n"), FALSE);
+    switch (biton32(default_layer_state))
+    {
+        case _COLEMAK:
+            oled_write_P(PSTR("Clmak"), FALSE);
+            break;
+
+        case _DVORAK:
+            oled_write_P(PSTR("Dvak"), FALSE);
+            break;
+
+        case _QWERTY:
+            oled_write_P(PSTR("Qwty"), FALSE);
+            break;
+
+        default:
+            oled_write_ln_P(PSTR("Undef"), FALSE);
+            break;
+    }
+#else
+    oled_write_P(PSTR("Layout: "), FALSE);
+    switch (biton32(default_layer_state))
+    {
+        case _COLEMAK:
+            oled_write_P(PSTR("Colemak"), FALSE);
+            break;
+
+        case _DVORAK:
+            oled_write_P(PSTR("Dvorak"), FALSE);
+            break;
+
+        case _QWERTY:
+            oled_write_P(PSTR("Qwerty"), FALSE);
+            break;
+
+        default:
+            oled_write_ln_P(PSTR("Undefined"), FALSE);
+            break;
+    }
+#endif
+}
+
+/** @brief displays the current active layer.
+ *
+ *  If any argument is invalid, the function has no effect.
+ *
+ *  @param void.
+ *  @return void.
+ */
+void render_layer_state(void)
+{
+    bool lower;
+    bool raise;
+    bool adjust;
+    bool numpad;
+
+    lower = layer_state_is(_LOWER) & !layer_state_is(_ADJUST);
```

Comment:
You could probably just use the `get_highest_layer` check and a switch to do this.

Reply:
Awesome, thanks for the input!